### PR TITLE
New Resources: aws_fsx_lustre_file_system and aws_fsx_windows_file_system

### LINKED ISSUE #7035: Amazon FSx Support
Amazon FSx is a service which was launched during AWS re:Invent 2018. They announced two different services as part of FSx: [Amazon FSx for Windows File Server](https://aws.amazon.com/blogs/aws/new-amazon-fsx-for-windows-file-server-fast-fully-managed-and-secure/) and [Amazon FSx for Lustre](https://aws.amazon.com/blogs/aws/new-amazon-fsx-for-lustre/).

Both services allow AWS users to create network shares using either [Windows File Server (SMB)](https://aws.amazon.com/fsx/windows/) or [Lustre](https://aws.amazon.com/fsx/lustre/). 

I was unable to find any resources associated with FSx in the official Terraform AWS provider documentation. Additionally, I was unable to find either a pull requests or issues related to FSx. I was wondering if somebody is working on Terraform AWS resources for FSx? 

I do not have an imitate need for FSx, that is the reason why I did not create a feature request.

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #7035
Closes #7074

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* **New Resource:** `aws_fsx_lustre_file_system`
* **New Resource:** `aws_fsx_windows_file_system`
```

Builds on #7074 with the following changes (special thanks to @slapula who put an incredible amount of work up until this point):

- Split into two separate resources (`aws_fsx_lustre_file_system` and `aws_fsx_windows_file_system`) to simplify practitioner configuration (e.g. lose the required configuration blocks), ease maintenance for resource logic and single file system arguments (e.g. `kms_key_id` only for Windows), and enhance validations (e.g. `storage_capacity` differences between file systems)
- Add covering acceptance testing with updates/force new for all arguments
- Better alignment with API naming for some arguments (e.g. changing `capacity` into `storage_capacity` to remove ambiguity with `throughput_capacity`)
- Implement [resource Customizable Timeouts](https://www.terraform.io/docs/extend/resources/retries-and-customizable-timeouts.html)
- Implement `skip_final_backup` argument for Windows
- Implement `network_interface_ids`, `owner_id`, and `vpc_id` attributes
- Augment resource documentation with additional notes found in [API reference](https://docs.aws.amazon.com/fsx/latest/APIReference/) and clarify `security_group_ids` import behavior

Output from acceptance testing:

```
--- PASS: TestAccAWSFsxLustreFileSystem_Tags (458.62s)
--- PASS: TestAccAWSFsxLustreFileSystem_basic (526.79s)
--- PASS: TestAccAWSFsxLustreFileSystem_disappears (564.58s)
--- PASS: TestAccAWSFsxLustreFileSystem_WeeklyMaintenanceStartTime (697.83s)
--- PASS: TestAccAWSFsxLustreFileSystem_StorageCapacity (940.13s)
--- PASS: TestAccAWSFsxLustreFileSystem_SecurityGroupIds (1074.08s)
--- PASS: TestAccAWSFsxLustreFileSystem_ImportedFileChunkSize (1276.18s)
--- PASS: TestAccAWSFsxLustreFileSystem_ImportPath (1322.13s)
--- PASS: TestAccAWSFsxLustreFileSystem_ExportPath (1338.75s)

--- PASS: TestAccAWSFsxWindowsFileSystem_Tags (2256.64s)
--- PASS: TestAccAWSFsxWindowsFileSystem_WeeklyMaintenanceStartTime (2346.44s)
--- PASS: TestAccAWSFsxWindowsFileSystem_basic (2404.35s)
--- PASS: TestAccAWSFsxWindowsFileSystem_disappears (2412.10s)
--- PASS: TestAccAWSFsxWindowsFileSystem_AutomaticBackupRetentionDays (2514.61s)
--- PASS: TestAccAWSFsxWindowsFileSystem_DailyAutomaticBackupStartTime (2849.43s)
--- PASS: TestAccAWSFsxWindowsFileSystem_SelfManagedActiveDirectory (2850.64s)
--- PASS: TestAccAWSFsxWindowsFileSystem_ThroughputCapacity (3292.44s)
--- PASS: TestAccAWSFsxWindowsFileSystem_SecurityGroupIds (3609.37s)
--- PASS: TestAccAWSFsxWindowsFileSystem_KmsKeyId (3977.10s)
--- PASS: TestAccAWSFsxWindowsFileSystem_StorageCapacity (4026.67s)
--- PASS: TestAccAWSFsxWindowsFileSystem_CopyTagsToBackups (4389.57s)
```

### CODE REVIEW

File: website/docs/r/fsx_lustre_file_system.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+* `create` - (Default `30m`) How long to wait for the file system to be created.
+* `delete` - (Default `30m`) How long to wait for the file system to be deleted.
+
+## Import
+
+FSx File Systems can be imported using the `id`, e.g.
+
+```
+$ terraform import aws_fsx_windows_file_system.example fs-543ab12b1ca672f33
```
Comment (ryndaniels) [inline website/docs/r/fsx_lustre_file_system.html.markdown]:
```suggestion
$ terraform import aws_fsx_lustre_file_system.example fs-543ab12b1ca672f33
```

**Comment by slapula** [comment]
Thanks for the help getting this over the finish line.  This approach makes a lot of sense and I completely forgot about custom timeout stuff.  This is awesome! (And it frees me up to work on adding support for Ground Station)

**Comment by bflad** [comment]
@slapula thank you so much for your work on this! ❤️ Would not have been possible without your support. My apologies that we wanted to get this released a little sooner than going through the various items above with you, which might have been challenging to convey via one-off review comments. We are discussing setting up some other forms of more real-time collaboration for this project in general to help solve this problem for development questions and more complicated review items (e.g. community office hours and/or provider development Slack channel). We would love to know if you would potentially be interested in something like that (if not, no big deal either). 😄 

If you have any questions about anything here, please reach out and I'll be happy to answer. Thanks again for making this happen.

**Comment by slapula** [comment]
@bflad Yes, I would definitely be interested!  I use Slack regularly so that's my preferred option but I'm open to other venues as well.  I'll be at Hashiconf so perhaps we can discuss this in person.

**Comment by onur-sam-gtn-ai** [comment]
It seems like `import` has let me down.

``` 
# aws_fsx_lustre_file_system.lustre[0] must be replaced
-/+ resource "aws_fsx_lustre_file_system" "lustre" {
      ~ arn                           = "redacted" -> (known after apply)
      ~ dns_name                      = "redacted.amazonaws.com" -> (known after apply)
        export_path                   = "s3://redacted"
      ~ id                            = "fs-redacted" -> (known after apply)
        import_path                   = "s3://redacted"
      ~ imported_file_chunk_size      = 2048 -> (known after apply)
      ~ network_interface_ids         = [
          - "eni-redacted",
          - "eni-redacted",
        ] -> (known after apply)
      ~ owner_id                      = "redacted" -> (known after apply)
      + security_group_ids            = [
          + "SAME SG ID AS THE ONE ALREADY ON THE FSx",
        ] # forces replacement
```

**Comment by onur-sam-gtn-ai** [comment]
State file indeed says `"security_group_ids": null`

**Comment by onur-sam-gtn-ai** [comment]
Raised issue #9795 for this.

**Pull request merged**